File: NetCore/ConsoleOop/ConsoleOop/Handlers/DbHandlers/IDBHandler.cs
using MyBackupCandidate;

namespace ConsoleOop.Handlers.DbHandlers
{
    /// <summary>
    /// 資料庫介面
    /// </summary>
    interface IDBHandler
    {
        /// <summary>
        /// 開啟資料庫連線
        /// </summary>
        void OpenConnection();

        /// <summary>
        /// 關閉資料庫連線
        /// </summary>
        void CloseConnection();

        /// <summary>
        /// 備份到資料庫處理
        /// </summary>
        /// <param name="candidate">檔案資訊</param>
        /// <param name="target">檔案內容</param>
        /// <returns>檔案內容</returns>
        byte[] Perform(Candidate candidate, byte[] target);
    }
}

File: Laravel/app/Services/Handlers/DBHandlers/IDBHandler.php
<?php
/**
 * Created by PhpStorm.
 * User: tsao
 * Date: 2017/11/25
 * Time: 下午 10:49
 */

namespace App\Services\Handlers\DBHandlers;


use App\Services\Configs\Candidate;

/**
 * 資料庫介面
 * Interface IDBHandler
 * @author steve.tsao
 * @package App\Services\Handlers\DBHandlers
 */
interface IDBHandler
{
    /**
     * 開啟資料庫連線
     * @author steve.tsao
     */
    function OpenConnection();

    /**
     * 關閉資料庫連線
     * @author steve.tsao
     */
    function CloseConnection();

    /**
     * 備份到資料庫處理
     * @author steve.tsao
     * @param Candidate $candidate 檔案資訊
     * @param string|null $target 檔案內容
     * @return string 檔案內容
     */
    function Perform(Candidate $candidate, $target): string;
}
